src/typings-new-format/spec-1.0/api.ts

```diff
@@ -118,31 +117,31 @@ export interface Overlay extends Api10 {
     usage?: string
 }
 
-export interface LibraryBase10 {
+export interface LibraryBase10 extends common.FragmentDeclaration {
 
     /**
      * Declarations of (data) types for use within this API
      */
-    types?: { [key: string]: datamodel.TypeDeclaration }
+    types?: datamodel.TypeDeclaration[]
 
     /**
      * Declarations of resource types for use within this API
      */
-    resourceTypes?: { [key: string]: resources.ResourceType10 }
+    resourceTypes?: resources.ResourceType10[]
```

Comment:
Previously resource type declaration contained the `name` property. I understand that the `key` has been moved to the `name`. Can this produce any conflict? Or was this always the same value? Though, the resource type itself does not contain a `name` property in RAML spec.